templates/devices/device_detail.html

```diff
@@ -622,6 +622,34 @@ <h4>{% trans "Installed software from puppetdb" %}</h4>
         });
     });
 
+    $(document).on('click', '.rotate-picture-right', function (e) {
+        console.log(e.toElement)
+        var picture_id = e.target.id.replace("picture", "")
+        var url = "{% url "device-api-picture-rotate" device.pk 0 "right" %}".replace("pictures/0", "pictures/"+picture_id)
+        $.ajax({
+            url: url,
+            type: 'PATCH',
+            success: function() {
+                alert("Picture rotated");
+
+            },
+        });
+    });
+
+    $(document).on('click', '.rotate-picture-left', function (e) {
+        console.log(e.toElement)
+        var picture_id = e.target.id.replace("picture", "")
+        var url = "{% url "device-api-picture-rotate" device.pk 0 "left" %}".replace("pictures/0", "pictures/"+picture_id)
+        $.ajax({
+            url: url,
+            type: 'PATCH',
+            success: function() {
+                alert("Picture rotated");
+
+            },
+        });
+    });
+
```

Comment:
Can you remove the duplication and just call a common function and pass the orientation to it?

As a useful success function we could reload the image on the fly. Possibly like this:

    success: function() {
        d = new Date();
        img = some.path.starting.from.e;
        img.src = img.src + '?' + d.getTime();
    }

Appending a timestamp to the image tricks the browser to ignore the cached image and we should immediately see the new, rotated image.

Reply:
Hmmm but this is an on click event and I have two buttons... I think that I would have to change the whole event type...
But this would only work if there was a redirect first, right? I tried using a HttpResponseRedirect, but that resulted in an error...

Reply:
We are in browser land here. jQuery is available and all we have to do for a reload is to modify the src attribute of the img tag.